File: Example Project/Test/ViewController4.m
//
//  ViewController4.m
//  Test
//
//  Created by Prince Ugwuh on 8/10/16.
//  Copyright © 2016 TTT. All rights reserved.
//

#import "ViewController4.h"
#import "Test-Swift.h"
#import "TestTextFieldCell.h"

@interface ViewController4 () <MSCollectionViewDataSourceDelegate>
@property(nonatomic, weak, nullable) IBOutlet UICollectionView *collectionView;
@property(nonatomic, strong, nullable) MSCollectionViewDataSource *dataSource;
@end

@implementation ViewController4

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.collectionView registerNib:[UINib nibWithNibName:[TestTextFieldCell identifier] bundle:nil] forCellWithReuseIdentifier:[TestTextFieldCell identifier]];
    
    [self build];
}


- (void)build {
    
    
    MSCollectionReusableViewItem *item = [[MSCollectionReusableViewItem alloc] init];
    item.identifier = [TestTextFieldCell identifier];
    item.item = nil;
    
    self.dataSource = [[MSCollectionViewDataSource alloc] initWithReusableViewItems:@[item] headerReusableViewItem:nil footerReusableViewItem:nil items:nil];
    self.dataSource.delegate = self;
    
    self.collectionView.dataSource = self.dataSource;
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

// Mark - MSTableViewDataSourceDelegate

- (void)dataSource:(TestTextFieldCell *)cell didConfigureCell:(NSIndexPath *)indexPath {
    cell.textFieldValueDidChange = ^(NSString *text){
        NSLog(@"%@", text);
    };
}

@end

File: Example Project Swift/Example Project Swift/ViewController4.swift
//
//  ViewController4.swift
//  Example Project Swift
//
//  Created by Prince Ugwuh on 10/12/16.
//  Copyright © 2016 MFX Studios. All rights reserved.
//

import UIKit

class ViewController4: UIViewController, CollectionViewDataSourceDelegate {
    @IBOutlet weak var collectionView: UICollectionView!
    
    var dataSource: CollectionViewDataSource?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.collectionView.register(UINib(nibName: TestTextFieldCell.identifier, bundle: nil), forCellWithReuseIdentifier: TestTextFieldCell.identifier)
        
        build()
    }
    
    func build() {
        
        let item = CollectionReusableViewItem()
        item.identifier = TestTextFieldCell.identifier
        
        self.dataSource = CollectionViewDataSource(reusableViewItems: [item])
        self.dataSource?.delegate = self
        self.collectionView?.dataSource = self.dataSource
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func didConfigure(_ cell: CollectionReusableViewDataSource, at indexPath: IndexPath) {
        let textFieldCell = cell as? TestTextFieldCell
        textFieldCell?.valueDidChange = { text in
            print(text)
        }
    }
}
